packages/pricing/src/services/pricing-module.ts

```diff
@@ -152,6 +153,79 @@ export default class PricingModuleService<
     return joinerConfig
   }
 
+  private setupCalculatedPriceConfig_(
+    filters,
+    config
+  ): PricingContext["context"] | undefined {
+    const fieldIdx = config.relations?.indexOf("calculated_price")
+    const shouldCalculatePrice = fieldIdx > -1
+    if (!shouldCalculatePrice) {
+      return
+    }
+
+    let pricingContext = filters.context ?? {}
+
+    // cleanup virtual field "calculated_price"
+    config.relations?.splice(fieldIdx, 1)
+    delete filters.context
+
+    return pricingContext
+  }
+
+  @InjectManager("baseRepository_")
+  async list(
+    filters: PricingTypes.FilterablePriceSetProps = {},
+    config: FindConfig<PricingTypes.PriceSetDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<PriceSetDTO[]> {
+    const pricingContext = this.setupCalculatedPriceConfig_(filters, config)
+
+    const priceSets = await super.list(filters, config, sharedContext)
+
+    if (pricingContext && priceSets.length) {
+      const calculatedPrices = await this.calculatePrices(
+        { id: priceSets.map((p) => p.id) },
+        { context: pricingContext },
+        sharedContext
+      )
+
+      for (let idx = 0; idx < priceSets.length; idx++) {
```

Comment:
we stopped using indexes to set because it was causing flakiness in the CI, might be better to create a map before and assign from that.

Reply:
In this case it is safe to do that because the method `calculatePrices` is returning the a `.map` of the input.
But in any case, I changed it to the map to keep the same pattern we use in other places.